src/garage/tf/policies/gaussian_gru_policy_with_model.py

```diff
@@ -179,7 +179,7 @@ def reset(self, dones=None):
             dones (numpy.ndarray): Bool that indicates terminal state(s).
 
         """
-        if not dones:
+        if dones is None:
```

Comment:
hmm why is `dones=None` an acceptable input for the RNN?

Reply:
I think it's more that `reset()` (without any parameters) is how the RNN is how the policy is restarted.
But `np.array` is mutable, so putting that as the default argument would be bad practice.

Reply:
Frustratingly, this only passed before because the `np.array` only had one element, otherwise it would have crashed with this exception:
```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
Should we make dones an argument without default value? Currently the experiment runner always provides dones as a numpy array.

Reply:
I would prefer to be able to call `reset()` without any arguments. It's fairly obvious what the input should be anyways.

Reply:
wow i wish there was a static analysis rule which would catch this.

@ahtsan are you sure that the `rollout` function doesn't call `reset()` without arguments?

Reply:
There's one hidden assumption though. When dones is None, it currently assigns dones to be np.array([True]), which implies vec_env=1.

Reply:
Perhaps it should be `np.repeat(True, self.vec_env)`? Either way, I don't actually know what the point of this argument is. I can't find any case in the codebase where we don't pass in an array of True values for this argument, but there are many places where we don't pass in anything.

Reply:
Ah, I see, I spoke too soon. We do actually pass this argument in the `OnPolicyVectorizedSampler`. I still think it's fine for it to have a default value, although I understand that this implies that the policy basically won't be "vectorized" then.

Reply:
I suggest we keep the current interface but state that policy won't be "vectorized" by default if dones is not passed.